### PR TITLE
GEOMESA-731 GeoMesaInputFormat - Allow number of mappers per shard to

### PR DESCRIPTION
**Pull request by aheyne**
... be configurable

* Added two new geomesa-site.xml config options:
** geomesa.mapreduce.split.count.absolute set the exact number of splits to use regardless of tserver count or accumulo-site.xml setting
** geomesa.mapreduce.split.count.tablet set the number of splits per tserver to use. Number of splits falls back to default accumulo splits computed by the job context.

### CODE REVIEW

**Comment by aheyne** [comment]
@elahrvivaz I have not tested this out yet.

File: docs/user/accumulo/configuration.rst [hunk 1]
```diff
@@ -30,6 +30,29 @@ geomesa.batchwriter.timeout.millis
 
 See the `Accumulo API <https://accumulo.apache.org/1.7/apidocs/org/apache/accumulo/core/client/BatchWriterConfig.html#setTimeout(long,%20java.util.concurrent.TimeUnit)>`__
 
+Map Reduce Splits Properties
```
Comment (elahrvivaz) [inline docs/user/accumulo/configuration.rst]: change this to 'Map Reduce Input Splits Properties' to be clearer

File: docs/user/accumulo/configuration.rst [hunk 2]
```diff
@@ -30,6 +30,29 @@ geomesa.batchwriter.timeout.millis
...
+----------------------------
+
+The following properties control the configure of Accumulo table splits.
```
Comment (elahrvivaz) [inline docs/user/accumulo/configuration.rst]: Change this to be 'control the number of input splits for a map reduce job'

File: docs/user/accumulo/configuration.rst (hunk 1 above)
Comment (aheyne) [inline docs/user/accumulo/configuration.rst]: Fixed

File: docs/user/accumulo/configuration.rst (hunk 2 above)
Comment (aheyne) [inline docs/user/accumulo/configuration.rst]: Fixed

**Comment by elahrvivaz** [comment]
@aheyne looks good - we need to test this out with and without the grouped splits, then we can merge - @jahhulbert-ccri might be able to help with that.

**Comment by jahhulbert-ccri** [comment]
i'll see if i can get some test cycles on it.

**Comment by jahhulbert-ccri** [comment]
starting testing.

File: geomesa-accumulo/geomesa-accumulo-jobs/src/main/scala/org/locationtech/geomesa/jobs/mapreduce/GeoMesaAccumuloInputFormat.scala [hunk 3]
```diff
@@ -164,13 +167,53 @@ class GeoMesaAccumuloInputFormat extends InputFormat[Text, SimpleFeature] with L
...
+          desiredSplits.toInt
+        } catch {
+          case e: java.lang.NumberFormatException =>
+            throw new IllegalArgumentException(s"Unable to parse geomesa.mapreduce.splits.tserver.max = $desiredSplits contains an invalid Int.", e)
+        }
+        if (numLocations > 0 && splitsPerTServer > 0) Some(numLocations * splitsPerTServer) else None
+      case None => None
+     }
+
+    grpSplitsMax.orElse(grpSplitsPerTServer) match {
```
Comment (jahhulbert-ccri) [inline geomesa-accumulo/geomesa-accumulo-jobs/src/main/scala/org/locationtech/geomesa/jobs/mapreduce/GeoMesaAccumuloInputFormat.scala]: Integer logic is messing this up. If you set max splits ==1 it works but not if you set max splits ==3 for instance when accumuloSplits =4

File: geomesa-accumulo/geomesa-accumulo-jobs/src/main/scala/org/locationtech/geomesa/jobs/mapreduce/GeoMesaAccumuloInputFormat.scala [hunk 4]
```diff
@@ -164,13 +167,53 @@ class GeoMesaAccumuloInputFormat extends InputFormat[Text, SimpleFeature] with L
...
+      case Some(numberOfSplits) =>
+        logger.debug(s"Using desired splits with result of $numberOfSplits splits")
+        val splitSize: Int = accumuloSplits.length / numberOfSplits
```
Comment (jahhulbert-ccri) [inline geomesa-accumulo/geomesa-accumulo-jobs/src/main/scala/org/locationtech/geomesa/jobs/mapreduce/GeoMesaAccumuloInputFormat.scala]: FYI look at this.... lets say ```accumuloSplits.length``` is ```4``` and try to set ```numberOfSplits =  1,2,3,3,4```

the split size = 
```
scala> 4/1
res0: Int = 4

scala> 4/2
res1: Int = 2

scala> 4/3
res2: Int = 1

scala> 4/4
res3: Int = 1

scala> 4/5
res4: Int = 0

```
If it is 5 in the config and 4 splits from accumulo it won't work:
```Exception in thread "main" java.lang.IllegalArgumentException: requirement failed: size=0 and step=0, but both must be positive```

**Comment by jahhulbert-ccri** [comment]
FYI test rig is:

``` 
find /tmp/geomesa-tools-path/lib/ -type f | tr '\n' ':' > /tmp/foo
HADOOP_OPTS="-Dgeomesa.mapreduce.splits.max=5  -agentlib:jdwp=transport=dt_socket,server=y,suspend=y,address=9898 $HADOOP_OPTS" HADOOP_CLASSPATH="$(cat /tmp/foo):$HADOOP_CLASSPATH"        hadoop jar ~/gitsrc/geomesa/geomesa-accumulo/geomesa-accumulo-jobs/target/geomesa-accumulo-jobs_2.11-1.3.1-SNAPSHOT.jar org.locationtech.geomesa.jobs.interop.mapreduce.FeatureCountJob -libjars "$(cat /tmp/foo)"
```

**Comment by aheyne** [comment]
Tested and appears to be working.

**Comment by jahhulbert-ccri** [comment]
only thing is see is that the illegal arg doesn't seem to propogate up...

**Pull request merged**